### PR TITLE
✅ Adds Percy.io Visual Testing to CI

### PR DESCRIPTION
**Pull request by bdolly**
Runs visual regression testing on all builds and PRs

### Motivation

We want to make sure that code changes made to isolated components don't have adverse effects on the whole system. Visual testing complements our existing unit and integration tests to verify that our entire UI–including components–aren’t just functioning correctly, but are also free of visual bugs or regressions.

### Use Cases
On each commit we push to a PR we want to kickoff CI and [compare a visual diff of our new stories to the base branches stories](https://docs.percy.io/docs/baseline-picking-logic) to ensure we changed all we intended and nothing that we didn't. This wil also help to scale the design system to a federated model so developers and designers alike can contribute with the security of knowing their contributions are tested and follow standards. 

### API changes
add npm script `"snapshot": "yarn storybook:build && percy-storybook --widths=320,1280 --minimum_height=2000"`

### Implementation Notes
This adds several [percy specific classes](https://docs.percy.io/docs/percy-specific-css) to hide non-essentail elements from rendering and being tested in percy snapshots. 

```css
@media only percy {
  /* set min-width for our stories so that components that are fill their container don't collapse */
  .percy-min-width{
    min-width: 960px;
  }

  /* hide addon-info header, props table, and description from Percy test */
  #root > div > div > div > div:first-child,
  #root > div > div > div > div:last-child,
  .hide-in-tests{
    display: none !important;
  }
  
  .show-in-tests{display: inline-block !important;}
  /* show our color swatches a11y table to check for changes */
  .ColorSwatch--a11y.hidden{display: block !important;}
}
```

### Rendering and Storybook location
Percy picks up and renders every story in Storybook by default. 
🚧  🚧 🚧 
I have attempted to hide some stories by using the percy story parameter `{ percy: {skip: true} }` on individual stories. However, this has not worked thus far and will be addressed with an upgrade to storybook v5 and percy v3
🚧  🚧 🚧 

### Functional Tests
Special test cases for percy only:
- render all button props combinations 
- render color swatches a11y tables open to check for changes 
- set header and card stories to a min-width of 960px in percy tests so they don't collapse

### CODE REVIEW

**Comment by dankolbman** [comment]
UIKit Storybook for Review

Built with commit 24a30e85c53aa662dddc9a12d27b020761b92fa7

https://deploy-preview-98--kf-uikit.netlify.com

File: README.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
   <a href="https://kf-uikit.netlify.com"><img src="https://img.shields.io/badge/storybook-master-ff69b4.svg?style=for-the-badge"></a>
   <a href="https://circleci.com/gh/kids-first/kf-uikit/tree/master"><img src="https://img.shields.io/circleci/project/github/kids-first/kf-uikit/master.svg?style=for-the-badge"></a>
 </p>
+[![This project is using Percy.io for visual regression testing.](https://percy.io/static/images/percy-badge.svg)](https://percy.io)
```
Comment (dankolbman) [inline README.md]: Can you use a shields.io badge?

Comment (bdolly) [reply README.md]: that's just a 💩 commit for now to see if the percy PR integration is working. Will remove in a rebase before this gets merged

**Comment by bdolly** [comment]
> Looks neat. I have a feeling we won't have a problem hitting the max snapshots per-month if we are actively developing, though.

paying for it would allow us to do mobile and IE regression testing though and save alot of worry on that front

File: README.md (hunk 1 above)
Comment (bdolly) [reply README.md]: 🔥

**Pull request merged**